File: Frontend/src/interfaces/v12/clientCode/clientCode.ts
import { CreatedChangedEntity } from "interfaces/v12/CreatedChangedEntity";
import { guid } from "types/guid";
import { PagedRequestAbstractDto } from "../pagedRequest/pagedRequestAbstractDto";

export interface ClientCodeDetailsDto extends CreatedChangedEntity {
  id: guid;
  name: string;
}

export interface ClientCodeCreateUpdateDto {
  name: string;
}

export interface ClientCodeSearchDto extends PagedRequestAbstractDto {
  search: string;
}

File: Wms.Api/Dto/ClientCode/ClientCode.cs
using Wms.Api.Entities;

namespace Wms.Api.Dto.ClientCode;

public class ClientCode
{
    public class ClientCodeDetailsDto : CreatedChangedEntity
    {
        public Guid Id { get; set; }
        public required string Name { get; set; }
    }

    public class ClientCodeCreateUpdateDto
    {
        public required string Name { get; set; }
    }

    public class ClientCodeSearchDto : PagedRequestAbstractDto
    {
        public required string Search { get; set; }
    }
}